### PR TITLE
improves techwebs a bit

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
experimental: nodes now make more sense
:cl:

removed redunant nodes from some nodes preq, lowered price on really useless nodes, mech guns and weapons nodes now respect standards

plasma cutters are in  a different node than adv plasma cutters
nutriment plus pump is in a different node than the basic pump
switched xray with beam lasers in adv beam unlock
lowered price of useless nodes: the one that unlocks tablets and the one that unlocls databases
temp gun is now under beam guns
emp gun is under electric weap
mech guns are now in the right place
put gps in basic bluespace
moved welding gas mask
removed diagnostic glasses from industrialed engineering bcs u unlock them somewhere else
new:
lowered price of huds amd increased night vision (no overall price change) bcs integrated huds are rpumdstart shit why would nt have a node for something that already exist?

same for cyborg upgrades, i felt like no one ever does them so the basic modules (everything except vtec ) cost less and vtec costs more

### CODE REVIEW

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -222,7 +222,7 @@
 	id = "posibrain"
 	display_name = "Positronic Brain"
 	description = "Applied usage of neural technology allowing for autonomous AI units based on special metallic cubes with conductive and processing circuits."
-	prereq_ids = list("mmi", "neural_programming")
```
Comment (XDTM) [inline code/modules/research/techweb/all_nodes.dm]: i think this is justified

File: code/modules/research/techweb/all_nodes.dm [hunk 2]
```diff
@@ -320,17 +320,17 @@
 	display_name = "Computer Consoles"
 	description = "Computers and how they work."
 	prereq_ids = list("datatheory")
-	design_ids = list("cargo", "cargorequest", "stockexchange", "libraryconsole", "aifixer", "mining", "crewconsole", "comconsole", "idcardconsole", "operating", "seccamera")
-	research_cost = 2500
+	design_ids = list("cargo", "cargorequest", "stockexchange", "libraryconsole", "mining", "crewconsole", "comconsole", "idcardconsole", "operating", "seccamera")
```
Comment (XDTM) [inline code/modules/research/techweb/all_nodes.dm]: Put operating in biotech maybe

File: code/modules/research/techweb/all_nodes.dm [hunk 3]
```diff
@@ -551,7 +551,7 @@
 	display_name = "Beam Weaponry"
 	description = "Various basic beam weapons"
 	prereq_ids = list("adv_weaponry")
-	design_ids = list("beamrifle", "ioncarbine")
+	design_ids = list("temp_gun", "xray_laser")
```
Comment (XDTM) [inline code/modules/research/techweb/all_nodes.dm]: >xray
>basic

Comment (Tlaltecuhtli) [reply code/modules/research/techweb/all_nodes.dm]: xray now is really bad and never used gameplay wise

**Comment by AnturK** [comment]
Please include changes and reasoning in the OP

File: code/modules/research/techweb/all_nodes.dm [hunk 4]
```diff
@@ -222,7 +222,7 @@
 	id = "posibrain"
 	display_name = "Positronic Brain"
 	description = "Applied usage of neural technology allowing for autonomous AI units based on special metallic cubes with conductive and processing circuits."
-	prereq_ids = list("mmi", "neural_programming")
+	prereq_ids = list("neural_programming", "mmi")
```
Comment (Dorsisdwarf) [inline code/modules/research/techweb/all_nodes.dm]: why did you reverse the order here?

Comment (Tlaltecuhtli) [reply code/modules/research/techweb/all_nodes.dm]: doesnt change anything

Comment (Dorsisdwarf) [reply code/modules/research/techweb/all_nodes.dm]: I know, just curious

**Comment by kevinz000** [comment]
There's a proc named total_techweb_points at the bottom, run the game server and call it and give me the new value.

**Comment by ExcessiveUseOfCobblestone** [comment]
Revert the bluespace change, don’t kneejerk. I certainly haven’t had enough time to see how it plays out ingame.

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
Don't touch export prices please i'm going to remove them _eventually_. They don't have any use now if you want to do it yourself for epic gbp :p

File: code/modules/research/techweb/all_nodes.dm (hunk 3 above)
Comment (ShadowDeath6) [reply code/modules/research/techweb/all_nodes.dm]: doesn't xray still annihilate blobs

Comment (XDTM) [reply code/modules/research/techweb/all_nodes.dm]: they stop after a blob hit now iirc

File: code/modules/research/techweb/all_nodes.dm [hunk 5]
```diff
@@ -268,8 +268,8 @@
 	id = "ai"
 	display_name = "Artificial Intelligence"
 	description = "AI unit research."
-	prereq_ids = list("robotics", "neural_programming")
```
Comment (kevinz000) [inline code/modules/research/techweb/all_nodes.dm]: bad

File: code/modules/research/techweb/all_nodes.dm [hunk 6]
```diff
@@ -432,7 +432,7 @@
 	display_name = "Advanced Cybernetic Implants"
 	description = "Upgraded and more powerful cybernetic implants."
 	prereq_ids = list("neural_programming", "cyber_implants","integrated_HUDs")
-	design_ids = list("ci-toolset", "ci-surgery", "ci-reviver")
+	design_ids = list("ci-toolset", "ci-surgery", "ci-reviver", "ci-nutrimentplus")
```
Comment (kevinz000) [inline code/modules/research/techweb/all_nodes.dm]: nutrimentplus is shit

File: code/modules/research/techweb/all_nodes.dm [hunk 7]
```diff
@@ -450,8 +450,8 @@
 	id = "basic_mining"
 	display_name = "Mining Technology"
 	description = "Better than Efficiency V."
-	prereq_ids = list("engineering")
-	design_ids = list("drill", "superresonator", "triggermod", "damagemod", "cooldownmod", "rangemod", "ore_redemption", "mining_equipment_vendor", "cargoexpress")//e a r l y    g a  m e)
+	prereq_ids = list("engineering", "basic_plasma")
```
Comment (kevinz000) [inline code/modules/research/techweb/all_nodes.dm]: why

File: code/modules/research/techweb/all_nodes.dm [hunk 8]
```diff
@@ -450,8 +450,8 @@
 	id = "basic_mining"
 	display_name = "Mining Technology"
 	description = "Better than Efficiency V."
-	prereq_ids = list("engineering")
-	design_ids = list("drill", "superresonator", "triggermod", "damagemod", "cooldownmod", "rangemod", "ore_redemption", "mining_equipment_vendor", "cargoexpress")//e a r l y    g a  m e)
+	prereq_ids = list("engineering", "basic_plasma")
+	design_ids = list("drill", "superresonator", "triggermod", "damagemod", "cooldownmod", "rangemod", "ore_redemption", "mining_equipment_vendor", "cargoexpress", "plasmacutter")//e a r l y    g a  m e)
```
Comment (kevinz000) [inline code/modules/research/techweb/all_nodes.dm]: why

File: code/modules/research/techweb/all_nodes.dm (hunk 3 above)
Comment (kevinz000) [reply code/modules/research/techweb/all_nodes.dm]: fair fair

File: code/modules/research/techweb/all_nodes.dm [hunk 9]
```diff
@@ -700,7 +700,7 @@
 	id = "mecha_tools"
 	display_name = "Exosuit Weapon (LBX AC 10 \"Scattershot\")"
 	description = "An advanced piece of mech weaponry"
-	prereq_ids = list("mecha", "adv_weaponry", "ballistic_weapons")
```
Comment (kevinz000) [inline code/modules/research/techweb/all_nodes.dm]: scattershot is literally the most op weapon in terms of sustained damage.

File: code/modules/research/techweb/all_nodes.dm (hunk 6 above)
Comment (Tlaltecuhtli) [reply code/modules/research/techweb/all_nodes.dm]: whats the point of having x and better x on the same node? same for cutter

File: code/modules/research/techweb/all_nodes.dm (hunk 5 above)
Comment (Tlaltecuhtli) [reply code/modules/research/techweb/all_nodes.dm]: neural programming is already unlocked with biotech its redunant

File: code/modules/research/techweb/all_nodes.dm (hunk 7 above)
Comment (Tlaltecuhtli) [reply code/modules/research/techweb/all_nodes.dm]: plasma cutters

File: code/modules/research/techweb/all_nodes.dm (hunk 9 above)
Comment (Tlaltecuhtli) [reply code/modules/research/techweb/all_nodes.dm]: and so what, you want to put different prices on guns?

Comment (Tlaltecuhtli) [reply code/modules/research/techweb/all_nodes.dm]: ballistic weapons already has adv weaponry, its redunant

**Comment by Tlaltecuhtli** [comment]
i cant really test bcs i havent figured out why i cant admin in my own server

**Comment by ExcessiveUseOfCobblestone** [comment]
You have to set the text correctly and make sure you’re logged into byond before running the server

File: code/modules/research/techweb/all_nodes.dm (hunk 6 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/research/techweb/all_nodes.dm]: agreed

File: code/modules/research/techweb/all_nodes.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/research/techweb/all_nodes.dm]: @ShadowDeath6 they are just lasers that give you rads and shoot through walls now.

File: code/modules/research/techweb/all_nodes.dm [hunk 10]
```diff
@@ -140,15 +140,15 @@
 	prereq_ids = list("practical_bluespace", "high_efficiency")
 	design_ids = list("bluespace_matter_bin", "femto_mani", "triphasic_scanning", "tele_station", "tele_hub", "quantumpad", "launchpad", "launchpad_console",
 	"teleconsole", "bag_holding", "bluespace_crystal", "wormholeprojector", "bluespace_pod")
-	research_cost = 15000
+	research_cost = 10000
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/research/techweb/all_nodes.dm]: No.

File: code/modules/research/techweb/all_nodes.dm (hunk 4 above)
Comment (fludd12) [reply code/modules/research/techweb/all_nodes.dm]: He had previously removed the prerequisites, but readded them in this commit by suggestion, that's why.

To be honest, the order kinda makes a little more sense this way, anyway, IMO.

File: code/modules/research/techweb/all_nodes.dm (hunk 9 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/research/techweb/all_nodes.dm]: knowing how techwebs work more than the author is concerning :^)

**Review by vuonojenmustaturska — CHANGES_REQUESTED** [review]
image

File: code/modules/research/techweb/all_nodes.dm (hunk 1 above)
Comment (KorPhaeron) [reply code/modules/research/techweb/all_nodes.dm]: Yeah, what is the rationale here?

**Comment by Tlaltecuhtli** [comment]
changes in op @AnturK @vuonojenmustaturska

File: code/modules/research/techweb/all_nodes.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/research/techweb/all_nodes.dm]: look at this in side-by-side mode @XDTM @KorPhaeron

File: code/modules/research/techweb/all_nodes.dm (hunk 8 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/research/techweb/all_nodes.dm]: early game build, you buy them after boots and Attack Speed stack

File: code/modules/research/techweb/all_nodes.dm [hunk 11]
```diff
@@ -460,7 +460,7 @@
 	display_name = "Advanced Mining Technology"
 	description = "Efficiency Level 127"	//dumb mc references
 	prereq_ids = list("basic_mining", "adv_engi", "adv_power", "adv_plasma")
-	design_ids = list("drill_diamond", "jackhammer", "hypermod", "plasmacutter", "plasmacutter_adv")
+	design_ids = list("drill_diamond", "jackhammer", "hypermod", "plasmacutter_adv")
 	research_cost = 2500
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/research/techweb/all_nodes.dm]: I know you didn't change this but this is fairly cheap considering adv_plassies make mining trivial

File: code/modules/research/techweb/all_nodes.dm (hunk 1 above)
Comment (XDTM) [reply code/modules/research/techweb/all_nodes.dm]: When i made the comment he had removed mmi, now it has been readded in reverse, so the diff does not appear outdated

File: code/modules/research/techweb/all_nodes.dm (hunk 11 above)
Comment (Tlaltecuhtli) [reply code/modules/research/techweb/all_nodes.dm]: ???

Comment (ExcessiveUseOfCobblestone) [reply code/modules/research/techweb/all_nodes.dm]: Laymen terms: 2.5K for Advanced Plasma Cutters???

File: code/modules/research/techweb/all_nodes.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/research/techweb/all_nodes.dm]: hacked

File: code/modules/research/techweb/all_nodes.dm [hunk 12]
```diff
@@ -342,16 +342,16 @@
...
 	research_cost = 1000
-	export_price = 5000
+	export_price = 2000
 
 /datum/techweb_node/comp_recordkeeping
 	id = "comp_recordkeeping"
 	display_name = "Computerized Recordkeeping"
 	description = "Organized record databases and how they're used."
 	prereq_ids = list("comptech")
 	design_ids = list("secdata", "med_data", "prisonmanage", "vendor", "automated_announcement")
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/research/techweb/all_nodes.dm]: NVG Prereq but didn't raise price on NVGs

**Comment by ExcessiveUseOfCobblestone** [comment]
are you trying to throw a curveball around techwebs on purpose?

File: code/modules/research/techweb/all_nodes.dm (hunk 12 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/research/techweb/all_nodes.dm]: changed in 80cbf83

**Review by ExcessiveUseOfCobblestone — APPROVED** [review]
God these feel so arbitrary, hopefully someone will make a standard for pricing

**Comment by davethwave** [comment]
A suggestion but how about putting the rapid syringe gun in ballistic weapons?

**Comment by Tlaltecuhtli** [comment]
my idea is lower prices on "low" tech stuff and higher on endgame of the tree branch to improve multi branch research over beeline endgame on specific ones.
if u want to suggest different prices without nerfing then speak
@davethwave the syringe gun is under medical weaps node

**Comment by davethwave** [comment]
I know but unless its 40 minutes in ingame no scientist would want to research such a thing...

**Comment by ExcessiveUseOfCobblestone** [comment]
> he thinks lower prices will stop beelining the meta web-route

I wish it were that easy.

**Comment by Tlaltecuhtli** [comment]
maybe put balistic and medical weaps nodes require weaponry instead of adv weaponry?

**Comment by ExcessiveUseOfCobblestone** [comment]
Let's not Joan techwebs, get this merged and see how it plays out.

**Comment by Tlaltecuhtli** [comment]
@anturk hey the reasoning is in op

**Pull request merged**